feat(core): add commandExternalDependencies hash input (#16916)
nrwl/nx: docs/shared/reference/project-configuration.md

Note the result value is hashed, so it is never displayed.

_External Dependencies_

> This input is useful for ambiguous executors like `nx:run-commands`.

Reviewer: This needs more explanation.  Something like

```suggestion
This input is useful for executors that do not have code directly associated with them, like `nx:run-commands`.
```

Reviewer: Maybe that's not the right explanation, but whatever the right explanation is needs to go here.

Examples:

- Specify that a target executor is not affected by any changes in installed packages

> - `{externalDependencies: []}`

Reviewer 2: ```suggestion
- `{externalDependencies: []}`
  - By default, to err on the side of caution, Specify that a target executor is not affected by any changes in installed packages
```

- Specify exact packages that affect a target executor
- `{externalDependencies: ["lerna"]}`

> Note the result value is hashed, so it is never displayed.

Reviewer: I don't know what this means

> Note the result value is hashed, so it is never displayed.

Reviewer 2: ```suggestion
This input type should only be used with executors which run arbitrary commands/scripts without having a way of determining which, if any, external dependencies are associated with the target. A canonical example is the `nx:run-commands`. Because any command can be provided as an option, it is not possible to tell what the command does or what external dependencies it may use.

> Note: By default, Nx will hash _all_ external dependencies for targets that use `nx:run-commmands` and other unknown executors. This is the safest option because it ensures that any external dependencies are captured. This may lead to unexpected cache misses when updating dependencies. This option is a means to override this behavior and allow for more cache hits by providing exactly which dependencies are used by the target.

Examples:
- Specify that a target executor is not affected by any changes in installed packages
- `{externalDependencies: []}`
- Specify exact packages that affect a target executor
- `{externalDependencies: ["lerna"]}`

Note the result value is hashed, so it is never displayed.
```
